labs/memory_bound/data_packing/validate.cpp

```diff
@@ -3,6 +3,49 @@
 #include <algorithm>
 #include <iostream>
 
+template <typename Received_t, typename Expected_t>
+static void report_failure(const char *var_name, Received_t received,
+                           Expected_t expected, int first_value,
+                           int second_value) {
+  std::cerr << "Validation Failed. Value " << var_name << " is " << received
+            << ". Expected is " << expected << " for intialization values "
+            << first_value << " and " << second_value << std::endl;
+}
+
+bool check_entry(int first_value, int second_value) {
+  S entry = create_entry(first_value, second_value);
+
+  if (entry.i != first_value) {
+    report_failure("i", entry.i, first_value, first_value, second_value);
```

Comment:
minor: maybe we can avoid code duplication...

Reply:
Agreed. Will change.